### PR TITLE
Seventh task

### PR DESCRIPTION
**Pull request by synion**
@kovson 
I don't have idea how to test this method in User

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -18,4 +22,5 @@ def destroy
   def user
     @user ||= User.find(params[:id])
   end
+
```
Comment (kovson) [inline app/controllers/users_controller.rb]: ✂️

File: app/models/user.rb [hunk 2]
```diff
@@ -1,11 +1,15 @@
+require 'csv'
```
Comment (kovson) [inline app/models/user.rb]: Do we really need to include that?

File: app/services/users_csv_export.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  private
+
+  def user_column(user)
+    attribute_columns(user) + interest_column(user)
+  end
+
+  def attribute_columns(user)
+    ATTRIBUTES.map { |column| user.send(column) }
```
Comment (kovson) [inline app/services/users_csv_export.rb]: Please do not user send (it is mainly used to access private method that shouldn't be accessed).

File: app/views/users/index.html.haml [hunk 4]
```diff
@@ -1 +1,2 @@
 = render users
+=link_to "CSV", users_path(format: :csv, locals: { users: users })
```
Comment (kovson) [inline app/views/users/index.html.haml]: space between link and =

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -17,5 +17,14 @@
     let(:user) { User.create_by_admin(email: 'lola@o2.pl')}
     it { user.valid_password?('secret').eql? true }
   end
+
+  describe '#to_csv' do
```
Comment (kovson) [inline spec/models/user_spec.rb:21]: This particular test has no value, you are just testing class of the output. 

You took a very nice approach to test service called by this method, so you already know that output is correct.

Please tell me what is this method doing in one sentence.

File: spec/services/users_csv_export_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+describe UsersCsvExport do
```
Comment (kovson) [inline spec/services/users_csv_export_spec.rb:3]: 👍 
Very nice.

**Comment by kovson** [comment]
Just few minors. Very nice job, specs are here, which is very good. We can also use feature spec to see that you can see button.

Please resolve conflicts.

@synion 👏

File: spec/models/user_spec.rb (hunk 5 above)
Comment (synion) [reply spec/models/user_spec.rb:21]: Is convert some collection of user to prepare for export to csv.

**Comment by synion** [comment]
@kovson  
I do features spec for Export but what i should test in model hmm?

**Comment by kovson** [comment]
@synion is User Model? I think that I already asked you about that in comment. What do you think this method ```to_csv``` is doing?

File: spec/models/user_spec.rb (hunk 5 above)
Comment (synion) [reply spec/models/user_spec.rb:21]: @kovson 
is convert some collection of user to prepare for export to csv.

**Review by synion — COMMENTED** [review]
@kovson
is convert some collection of user to prepare for export to csv.
sorry i don't mark you on last coment

File: spec/models/user_spec.rb (hunk 5 above)
Comment (kovson) [reply spec/models/user_spec.rb:21]: It method ```to_csv``` doing that or ```UsersCsvExport```?

Comment (synion) [reply spec/models/user_spec.rb:21]: UsersCsvExport
to_csv i use method generat on UsersCsvExport.new with collection of User.includes(:interests)

Comment (kovson) [reply spec/models/user_spec.rb:21]: But in fact ```to_csv``` is just calling ```UsersCsvExport``` and result from ```UsersCsvExport``` is already tested by specs for service. 

So the only thing that should be tested is - is ```UserCsvExport``` called with correct arguments?

https://relishapp.com/rspec/rspec-mocks/docs/basics/expecting-messages

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (kovson) [reply app/controllers/users_controller.rb]: ➖ ✂️

Comment (synion) [reply app/controllers/users_controller.rb]: sorry i do pull -f and i rest HEAD and i must do many thing second time

File: spec/models/user_spec.rb (hunk 5 above)
Comment (kovson) [reply spec/models/user_spec.rb:21]: @synion 
Look at example at link above. We should expect ```UserCsvExport``` to receive new and then generate.

**Comment by kovson** [comment]
@synion 
A lot of work has been done with that task. Very good job. Lasy two task should be easy, we can skip feature specs for both of them.

**Pull request merged**